DirectXTex/DirectXTexMipmaps.cpp

```diff
@@ -2521,6 +2521,129 @@ namespace
 // Entry-points
 //=====================================================================================
 
+float clamp(float value, float min, float max)
+{
+    return value < min ? min : value > max ? max : value;
+}
+
+//-------------------------------------------------------------------------------------
+// Alpha to coverage, for mip generation
+//-------------------------------------------------------------------------------------
+_Use_decl_annotations_
+float CalculateAlphaCoverage(const Image& image, const float alphaCoverageRatio, const float scale)
+{
+    float coverage = 0.0f;
+
+    ScopedAlignedArrayXMVECTOR scanline(reinterpret_cast<XMVECTOR*>(_aligned_malloc((sizeof(XMVECTOR) * image.width), 16)));
```

Comment:
``_aligned_malloc`` doesn't throw on failure, so you need to explicitly check for out-of-memory. If nothing else, it should silently fail right here to avoid a crash in out-of-memory scenarios.

    if (!scanline)
        return 0.f

A better approach would be to have the caller pre-allocated the scratch memory so you return an appropriate ``HRESULT`` failure on out-of-memory. You should also allocate the largest possible width and reuse it instead of current implementation which is allocating/freeing the memory over and over again.